### DISCUSSION TITLE
2024-09-27 - Autonomy and Approvals - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by bradystroud** [SSWConsulting/SSW.Rules.Content#9265] [Chewing the Fat]
https://www.ssw.com.au/rules/cross-approvals/

How do you like to get approvals?  (Teams app, email, SugarLearning etc.) 
Any other ideas to help scale approvals?

**Comment by tiagov8** [comment]
I never tried **Teams app**.

**Email** is bad...

**SugarLearning** is good, but I would love to automate some.   
E.g. Instead of someone checking...    
**Re: {{ NAME }} - How to use Teams Channels**     
OR    
**Re: {{ NAME }} - Induction Part 1 - Knowledge check - Changing X to Y**   
...we could make those as a quiz, so people can move on after they pass.

**Reply by jaydenalchin** [reply]
+1 - A lot of the SugarLearning approvals could be automated & streamlined

**Comment by drwharris** [comment]
Anything that keeps momentum and stops things like approvals moving offline, eg email, is good.

**Comment by danielmackay** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams approvals - it's super convenient.  On mobile I get a push notification and can approval in a couple of clicks.

> Any other ideas to help scale approvals?

No other ideas on top of what @ricksu978 and I presented in the Approvals Spec Review

**Comment by TobyChurches** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Generally, I like to message or call via teams, Followed by a email to document it - I find that it is generally a lot quicker

> Any other ideas to help scale approvals?

No new Ideas but the 'Teams Approvals app' looks promising

**Comment by leahy268** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams App would be best I think.

> Any other ideas to help scale approvals?

Top level approvals should only be for things that change the direction of the company.

**Comment by jakebayliss** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Typically dont need approvals too often so email works fine for me. Haven't tried teams for approvals but seems like it would work well at scale

**Comment by ozdemir-mehmet** [comment]
Team Approvals + Email

**Comment by tkapa** [comment]
I guess I'd prefer it to be over an email, for the tracking. But if Teams app has tracking available it should be fine.

We mostly need these approval groups because people have an overflow of responsibilities. If many (more than 3-4) people are needed, you likely need to look at the process.

**Comment by Tanya-Leahy** [comment]
>How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams App sounds great. 
>Any other ideas to help scale approvals?

No I can't think of any new ideas, but the Teams App is a great idea.

**Comment by kikibianc** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)
> Any other ideas to help scale approvals?

Teams app is infinitely customizable and the best option.

**Comment by joshbermanssw** [comment]
> How do you like to get approvals? 

When I can, i'll try to get the approval in person (or via Teams call) and then followed up with some email documentation 

> Any other ideas to help scale approvals

I like the look of the Teams Approval customizability

**Comment by ncn-ssw** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Email - at the moment. The Teams Approvals app idea looks good.

> Any other ideas to help scale approvals?

+1 for @tkapa's comment - "We mostly need these approval groups because people have an overflow of responsibilities. If many (more than 3-4) people are needed, you likely need to look at the process. "

**Comment by ricksu978** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

I have no preference as a person getting approvals, as long as the process works.
However, from approvers' perspective, Teams app is a good place for managing approvals.

> Any other ideas to help scale approvals?

Integrate AI pre-evaluation of the approval based on previous approvals.
For example, Salary Sacrifice, Mehmet always rejects 13" laptop in the past, with AI's assistance, it could pick this up early in the process and save time for both approvers and person getting the approval.

**Comment by Levijj22** [comment]
We played around with Teams Approvals when Brainstorming the Levi.AI tool. 
It has some limitations, but works great! 

+1 to move from emails to this app wherever possible

**Comment by edkranz** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams app, email.

> Any other ideas to help scale approvals?

Having approvals on a single platform makes sense, allows you to track things historically with ease.

**Comment by jimmy1992abc** [comment]
I played a lot with some Teams App 2 years ago, including Approvals. Ideally, everything should be integrated into Teams, by exploring the App marketplace or even creating apps just for ourselves.
Many admin bottlenecks are slowing SSW down, but I am so happy that we are starting to lift it.

**Comment by AttackOnMorty** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams app - Let's not reinvent the wheel if it can meet most of our needs.

**Comment by RobJT-wq** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

I think I'd like to get approvals in the Teams app, I feel like sending approvals in emails can get lost from time to time.

**Comment by JackDevAU** [comment]
A Teams app or an email could work fine for approvals. As long as people can easily approve it and it isn't forgotten 🙏

**Comment by steven0x51** [comment]
>How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams App is the best

**Comment by AntPolkanov** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams 🚀

**Comment by jeoffreyfischer** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams app - approvals should be quick and easy

**Comment by StefStar** [comment]
I like Teams app for approvals since its super straightforward and most of the time will send an email anyway as a backup.

**Comment by Calinator444** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

When receiving approvals, I prefer to get them through email or by sending an "as per our conversation" email. This creates a ledger to confirm the  approval that either party can respond to.

> Any other ideas to help scale approvals?

I would use the teams app

**Comment by chrisschultzssw** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Zendesk works well for SysAdmin related things, but I also like Teams Approvals.

**Comment by isaaclombardssw** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams sounds like the best option. The SugarLeaning approval process isn't bad either – but it could take some getting used to if we have it as something for both validating work / progress (current use) and making business choices.

**Comment by Ben0189** [comment]
>How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams and email

**Comment by stephanfakossw** [comment]
I think purchase approvals need work ⚠️

Previously, the accounting team researched options for purchase approvals but the problem was small credit card purchases - there is no technical solution for credit card purchase approvals. Credit cards presume budgets that are already pre-approved so there is no approval step when someone is buying a minor item on the credit card (I.e. Levi doesn't need to ask for approval to purchase coffee for clients, or to post material etc). Since then, we got pre-approvals for some small items, but the system now looks a little bit complex.

Purchase approvals in emails are just the worst way to go about purchases - it's hard to track. There's alternatives like Zendesk.

**Comment by uly1** [comment]
Ase an AI tool to approve easy ones... only pass on to a human approver for hairy cases that arent obviously fine to approve.
Maybe use YakShaver to ask for the approval and it will rout to the approver AI (i've added  PBI to yakshaver for that: https://github.com/SSWConsulting/SSW.YakShaver/issues/744)

**Comment by bettybondoc** [comment]
> How do you like to get approvals?

The Teams approvals look promising

> Any other ideas to help scale approvals?

Can't think of any at the moment

**Comment by SamBWagner** [comment]
How do you like to get approvals? (Teams app, email, SugarLearning etc.)

I pretty much universally just call someone and ask them. If yes, I do it. If it's a critical decision I'll document it in a PBI/email.

Any other ideas to help scale approvals?

I think having some sort of simple list, literally a notepad doc would be enough, of things people are "known" or "responsible for" would be great. This whole system is super over-cooked

**Comment by jaydenalchin** [comment]
Design work is (by design) iterative. It involves gathering feedback from multiple sources and rarely involves tick-and-flick approval. For this reason I prefer seeking approvals with an actual chat, discussion and review (asynchronously if needed) with decisions or feedback documented in the relevant PBI or thread. 

I start with peer-review and then seek approval from the relevant responsible person (i.e. a Product Owner). When the process gets more complex than this - or requires more people/input than this - is when we get stuck in Approval Hell.

**Reply by jaydenalchin** [reply]
SugarLearning approvals feel like its own issue with its own solutions (automation) and the process as it is now is certainly not scalable.

**Reply by bettybondoc** [reply]
+1

Design cross-approvals might be good:
- Social media/graphic design stuff = 2 Designers (OR 1 Marketing + 1 Designer)
- UI/UX = 2 Designers + PO

But essentially 2 designers for peer-review 🤝

**Comment by PennyWalker** [comment]
I would love a Teams App if it tracks everything for the accountants, we burn so much time tracking approvals after the fact that it's a nightmare for them. I've seen how hard Jimmy has to chase sometimes. 

Sugarlearning is awesome, it's easy and simple but it only allows 1 approver so we'd need to change that.

**Comment by jernejk** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)
Teams and/or forms.

> Any other ideas to help scale approvals?
In the last 15 years + the experience I have received from my parents managing 200+ people companies, emails don't scale and always result in more trouble than they solve, especially when there are more people.

When trying to understand how other consultants work, I noticed that we have a lot in common with self-employed consultants, where everyone is an isolated unit that needs to report separately from the team.

**Comment by babakamyljanovssw** [comment]
I also agree with most of one here about Teams app for approvals 🚀

**Comment by tino-liu** [comment]
Another vote on Teams app for approvals!

**Comment by Aibono1225** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams app looks promising

**Comment by jimmidier** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams app would be ideal at the moment

**Comment by bradystroud** [comment]
I like conversations.
I'd like to record approvals in the Approvals app, but it needs a lots of improvements to support more scenarios 😐

I think its also important to track the rejections (this data helps with alignment and training an AI)
It would be great if EagleEye did this but that would mean sending an email for every version of the email

e.g.
1. Employee sends v1
1. Bradys checks it and asks for changes
1. Employee sends v2
1. Brady approves
1. Piers checks it and asks for changes
1. Employee sends v3
1. Piers approves
1. Employee sends v4
1. Adam approves

sounds painful 👎 but it gets the data we need


Check > Fix > Send

**Reply by bradystroud** [reply]
Might make sense to do 'test please' emails instead of versions.

**Reply by bradystroud** [reply]
Moved this to a PBI so it doesnt get lost
https://github.com/SSWConsulting/SSW.EagleEye/issues/847

**Comment by jerryluossw** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams seems to be the most convenient.

**Comment by matt-goldman-ssw** [comment]
>How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams is good. You get the notification plus you can use adaptive cards to do it in-line.


>Any other ideas to help scale approvals?

Most of these things shouldn't need a human approver. If we have rules in place to define who can get what and when, the approval workflow can be automated. Then the human escalation is freed up for curly ones.

**Comment by miichaelsmedley** [comment]
> https://www.ssw.com.au/rules/cross-approvals/
> 
> How do you like to get approvals? (Teams app, email, SugarLearning etc.) 
email or Teams - either is fine. Email the approval after using Teams 

>Any other ideas to help scale approvals?
use both together.

**Comment by bahjatalaadelssw** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams approvals - Or Devops. I like having the PBI be the single source of truth.

> Any other ideas to help scale approvals?

Each team should have a lead that handles approvals. If they are in two minds, then talk to a state manager. Building accountability creates consistency.

**Comment by Marxoz** [comment]
I don't think changing the way we manage approvals will have an impact yet. We first need to sort out our approval processes and delicate responsibility and authority before looking at how we would capture them going forward.

**Comment by sam-s-maher** [comment]
I don't mind any system - but would prefer consistency. e.g The mixed process in induction of sending off emails & embedded approvals can sometimes feel inconsistent, which has an effect on comprehension and motivation.

**Comment by awaltos** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

(Assuming this means "How do I like to get other people's requests for my approval?")
Wherever it is easiest to see and review the request. For example, email or teams so I can get a notification and have that persist until I get the chance to action it. Next most important is to make it easy for me to approve/deny - Teams app looks good for this.

> Any other ideas to help scale approvals?

Automation and removing human where possible is best. Otherwise, multiple approvers help when scaling.

**Comment by MattParkerDev** [comment]
Cross approvals certainly alleviate being blocked by a sole approver.

**Comment by amankumarrr** [comment]
Teams approval sounds like a good option, as it offers a centralized platform where everyone in the company is expected to be active and can respond quickly.

**Comment by vladislav-kir** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams App sounds promising.

> Any other ideas to help scale approvals?

No other ideas, but excited to see what Teams App can improve.

**Comment by ozair-ashfaque** [comment]
> How do you like to get approvals? (Teams app, email, SugarLearning etc.)

Teams app - if this method is the quickest and most efficient

> Any other ideas to help scale approvals?

I have no other ideas.
